enterprise/internal/insights/background/historical_enqueuer.go

```diff
@@ -262,9 +262,27 @@ func (h *historicalEnqueuer) Handler(ctx context.Context) error {
 	if err := h.buildFrames(ctx, uniqueSeries, sortedSeriesIDs); err != nil {
 		return multierror.Append(multi, err)
 	}
+	if err == nil {
+		// we successfully performed a full repo iteration without any "hard" errors, so we will update the metadata
+		// of each insight series to reflect they have seen a full iteration. This does not mean they were necessarily successful,
+		// only that they had a chance to queue up queries for each repo.
+		h.markInsightsComplete(ctx, foundInsights)
+	}
+
 	return nil
 }
 
+func (h *historicalEnqueuer) markInsightsComplete(ctx context.Context, completed []itypes.InsightSeries) {
+	for _, series := range completed {
+		_, err := h.dataSeriesStore.StampBackfill(ctx, series)
+		if err != nil {
+			// do nothing to preserve at least once semantics
+			continue
+		}
+		log15.Info("Insight marked backfill complete.", "series_id", series.SeriesID)
```

Comment:
```suggestion
		log15.Info("insights: Insight marked backfill complete.", "series_id", series.SeriesID)
```

Does this still run in a shared service? If so, probably makes sense to prefix with a common string you can grep for.